### DISCUSSION TITLE
Remove static keyword from members of static classes

### DISCUSSION THREAD

**Discussion by osamakawish** [dotnet/csharplang#4248] [Q&A] [answered]
Since all the members of a static class must be static, is there a reason why the `static` keyword is still required and not just assumed? 

(This sort of fits under ideas as well as Q&A equally.)

**Comment by Unknown6656** [comment, marked as answer]
Probably because adding or removing `static` from `public [static] class MyClass {` would break the existing static methods.
Imagine the following:
```csharp
public static class MyClass
{
    int i = 24;
    
    static void M() => Console.WriteLine(i);
}
```
removing the `static`-keyword from the first line would break the following code (e.g. in an other class):
```csharp
MyClass.i = 999;
MyClass.M();
```

**Reply by 333fred** [reply]
Yep, this is the reasoning. It's to be more resilient to refactorings.

**Reply by osamakawish** [reply]
I tried this code:

```c#
public static class MyClass
{
    int i = 24;
    
    static void M() => Console.WriteLine(i);
}
```
and I get the same error as you would for trying to add a non-static member:

> cannot declare instance member in a static class

so I'm not sure what the issue here is, as this is already "illegal" in C# right now.

I could be wrong, but all members in a `static` class have to be marked `static`. Which raises the question as to why it's not inferred instead.

Or is that line supposed to be the following?:

```c#
    static int i = 24
```

**Reply by osamakawish** [reply]
Ok nvm, I think I get the gist of your point.

**Reply by osamakawish** [reply]
But now I guess my question is: "If we're going to mark everything `static` anyways, what's the point in declaring a class as `static`"? With the exception of maybe static generics.

**Reply by 333fred** [reply]
> "If we're going to mark everything static anyways, what's the point in declaring a class as static"

To prevent you from instanciating the class. It's not intended to be stateful, it's intended to be static.
